File: app/src/main/java/com/tommihirvonen/exifnotes/datastructures/FilterMode.java
package com.tommihirvonen.exifnotes.datastructures;

/**
 * Enum class to describe roll filter modes.
 */
public enum FilterMode {
    ACTIVE(0),
    ARCHIVED(1),
    ALL(2);

    int value;

    FilterMode(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static FilterMode fromValue(int value) {
        switch (value) {
            case 0: default:
                return ACTIVE;
            case 1:
                return ARCHIVED;
            case 2:
                return ALL;
        }
    }
}

File: app/src/main/java/com/tommihirvonen/exifnotes/datastructures/FilterMode.kt
package com.tommihirvonen.exifnotes.datastructures

/**
 * Enum class to describe roll filter modes.
 */
enum class FilterMode(value: Int) {
    ACTIVE(0),
    ARCHIVED(1),
    ALL(2);

    var value: Int = 0
        internal set

    init {
        this.value = value
    }

    companion object {

        fun fromValue(value: Int): FilterMode {
            return when (value) {
                0 -> ACTIVE
                1 -> ARCHIVED
                2 -> ALL
                else -> ACTIVE
            }
        }
    }
}
